src/Symfony/Component/AstGenerator/AstGeneratorInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * An AstGeneratorInterface is a contract to transform an object into an AST.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface AstGeneratorInterface
+{
+    /**
+     * Generate an object into an AST given a specific context.
+     *
+     * @param mixed $object  Object to generate AST from
+     * @param array $context Context for the generator
+     *
+     * @return \PhpParser\Node[] An array of statements (AST Node)
+     */
+    public function generate($object, array $context = array());
+
+    /**
+     * Check whether the given object is supported for generation by this generator.
+     *
+     * @param mixed $object Object to generate AST from
+     *
+     * @return bool
+     */
+    public function supportsGeneration($object);
```

Comment:
I don't really understand use-case of the chain Generator, to be honest. Probably, another abstraction is missing like `interface AstTreeWalker::walk(array $nodes): array` or something? 

For me, `AstGeneratorInterface` looks like the following generic interface:
```php
/**
 * Maps something on something another. Very useful abstraction.
 */
interface Mapper
{
    /**
     * @return mixed something
     */
    public function map($something);
}
```
It generates _some_ AST node of _some_ object. The purpose of this interface is very unclean. But some kind of `AstWalker` is probably makes sense if you have examples of generic ones (coding style fixers, dead code removers, etc.).